packages/zoe/src/contracts/loan/close.js

```diff
@@ -43,14 +43,18 @@ export const makeCloseLoanInvitation = (zcf, config) => {
     // Transfer the repaid loan amount to the lender
 
     repaySeat.incrementBy(
-      collateralSeat.decrementBy({
-        Collateral: collateralSeat.getAmountAllocated(
-          'Collateral',
-          collateralBrand,
-        ),
-      }),
+      collateralSeat.decrementBy(
+        harden({
+          Collateral: collateralSeat.getAmountAllocated(
+            'Collateral',
+            collateralBrand,
+          ),
+        }),
+      ),
+    );
+    lenderSeat.incrementBy(
+      harden(repaySeat.decrementBy(harden({ Loan: debt }))),
     );
```

Comment:
Ideally the creator of an object is the one to harden the object before releasing it. The typical and best case, as in the remaining harden here, is on the object literal itself. Otherwise, things should only be passing hardened objects to each other. The receiver should be defensively checking and rejecting, as `incrementBy` already does. It should not be doing its own defensive hardening, as that masks bugs in the source of the object. For example, if `decrementBy` returned an unhardened object, the defensive harden would mask the bug. In this case it is fine because `decrementBy` is not buggy in this way.
```suggestion
    lenderSeat.incrementBy(repaySeat.decrementBy(harden({ Loan: debt })));
```
Please correct all places where this PR makes this error.

This will also cause some error messages to revert to what they were before this PR. IMO the old error messages were better anyway, so this is welcome.

Reply:
Hmm, I'm not seeing how this will cause some error messages to revert. I would expect the effect to be the same.

But, I just removed all of the `harden` calls on outputs from `incrementBy` and `decrementBy`, since they are already hardened.